src/asciidoc/docs/develop/testingTemplates.adoc

```diff
@@ -0,0 +1,208 @@
+
+The *Remote* tab offers several additional ways of previewing your Koji.
+
+* On a mobile device – Open a preview on any device by scanning a QR code.
+Logs from the device appear in a pane in the editor.
+
+* In a new tab – Open a preview on a staging server.
+Logs from the tab appear in a pane in the editor.
+
+* In the *Live player* – Open another browser window where the Koji is displayed in an iframe, just as the published Koji will display.
+You can also use the *Live player* to test <<_testing_in_app_purchases,in-app purchases>>.
+Logs from the browser window appear in a pane in the editor.
+
+* With a shared link – You can copy the link for the preview.
+If you share this link with someone else, they can preview the Koji while it's still in development.
++
+[NOTE]
+The remote preview link will only work for as long as your editor session is still active, so it is not suitable for long-term sharing.
+
+=== Testing backend routes
+
+The *Routes* tab lets you test your backend routes.
+It's similar to using a program like curl or Fiddler.
+For example, you can manually enter an HTTP request, configure a header for it, send the request, and examine the results returned by the server.
+
+=== Writing to the console
+
+You can test JavaScript code by using the `console.log` method, which prints to the console.
+To see the console in the Koji editor, click the console button image:consoleButton.jpg[Console button] in the *Live preview* pane.
+
+For example, the following code snippet prints *Key pressed* to the console every time you press a key, as long as the *Live preview* pane is selected.
+
+[NOTE]
+You can select *Live preview* by clicking anywhere in its pane.
+
+[source,javascript]
+----
+    $(window).keydown(function (e) {
+        KEY_STATUS.keyDown = true;
+        if (KEY_CODES[e.keyCode]) {
+            e.preventDefault();
+            KEY_STATUS[KEY_CODES[e.keyCode]] = true;
+        }
+        console.log("Key pressed");
+    }).keyup(function (e) {
+        KEY_STATUS.keyDown = false;
+        if (KEY_CODES[e.keyCode]) {
+            e.preventDefault();
+            KEY_STATUS[KEY_CODES[e.keyCode]] = false;
+        }
+    });
+----
+
+When running your Koji outside the editor, you can access the console through the browser's menu.
+(See <<_using_browser_tools_for_testing>>.)
+
+=== Testing in-app purchases
+
+If your template supports <<withkoji-koji-iap-package#,in-app purchases>>, you can test the functionality by clicking *Debug in live player* on the *Remote* tab.
+
+. Log into your Koji account and click your profile icon to open the main menu.
+. Click *Wallet*.
+. If your *Wallet* is empty, top it up.
+. In another browser window, open the Koji editor.
+. Open the *Remote* tab and click *Debug in live player*.
+. Try paying yourself a small amount.
+. Go back to your *Wallet* to check that the transaction was successful.
+. If all went well, you'll see the amount you paid deducted from your balance.
```

Comment:
As mentioned in a [previous comment](https://github.com/madewithkoji/docs/pull/109#discussion_r538896371), this is not a step because it's not an action the user takes.